src/drivers/github.go

```diff
@@ -2,25 +2,42 @@ package drivers
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/Originate/git-town/src/git"
 )
 
-// GithubCodeHostingDriver provides tools for working with repositories
-// hosted on Github
-type GithubCodeHostingDriver struct{}
+type githubCodeHostingDriver struct {
+	originURL string
+	hostname  string
+}
+
+func (d *githubCodeHostingDriver) CanBeUsed() bool {
+	fmt.Println(d.hostname)
+	return d.hostname == "github.com" || strings.Contains(d.hostname, "github")
+}
 
-// GetNewPullRequestURL returns the URL of the page
-// to create a new pull request on Github
-func (driver *GithubCodeHostingDriver) GetNewPullRequestURL(repository string, branch string, parentBranch string) string {
+func (d *githubCodeHostingDriver) GetNewPullRequestURL(repository string, branch string, parentBranch string) string {
 	toCompare := branch
 	if parentBranch != git.GetMainBranch() {
 		toCompare = parentBranch + "..." + branch
 	}
 	return fmt.Sprintf("https://github.com/%s/compare/%s?expand=1", repository, toCompare)
 }
 
-// GetRepositoryURL returns the URL of the given repository on github.com
-func (driver *GithubCodeHostingDriver) GetRepositoryURL(repository string) string {
+func (d *githubCodeHostingDriver) GetRepositoryURL(repository string) string {
 	return "https://github.com/" + repository
 }
+
+func (d *githubCodeHostingDriver) HostingServiceName() string {
+	return "Github"
+}
+
+func (d *githubCodeHostingDriver) SetOriginURL(originURL string) {
```

Comment:
This was the easiest way to inject this dependency without a ton of duplication or global variables.